live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/storage/StorageClientIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageClient;
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageConfig;
+import org.alfresco.hxi_connector.live_ingester.utils.DockerTags;
+
+@SpringBootTest(classes = {LocalStorageConfig.class, SignedStorageClient.class})
+@ActiveProfiles({"test"})
+@Testcontainers
+class StorageClientIntegrationTest
+{
+    private static final String LOCALSTACK_IMAGE = "localstack/localstack";
+    private static final String LOCALSTACK_TAG = DockerTags.getOrDefault("localstack.tag", "3.0.2");
+    private static final String BUCKET_NAME = "test-hxinsight-bucket";
+    private static final String OBJECT_KEY = "dummy.txt";
+    private static final String OBJECT_CONTENT = "dummy's file dummy content";
+    private static final String OBJECT_CONTENT_TYPE = "plain/text";
+
+    @Container
+    @SuppressWarnings("PMD.FieldNamingConventions")
+    private static final LocalStackContainer localStack = new LocalStackContainer(DockerImageName.parse(LOCALSTACK_IMAGE).withTag(LOCALSTACK_TAG));
+
+    @Autowired
+    private LocalStorageClient testS3Storage;
+    @Autowired
+    private StorageClient storageClient;
+
+    @BeforeAll
+    static void beforeAll() throws IOException, InterruptedException
+    {
+        localStack.execInContainer("awslocal", "s3api", "create-bucket", "--bucket", BUCKET_NAME);
+    }
+
+    @Test
+    void testUpload() throws IOException
+    {
+        List<String> freshBucketContent = testS3Storage.listBucketContent(BUCKET_NAME);
+        assertThat(freshBucketContent)
+                .isNotNull()
+                .isEmpty();
+
+        // given
+        try (InputStream fileInputStream = new ByteArrayInputStream(OBJECT_CONTENT.getBytes()))
+        {
+            URL preSignedUrl = testS3Storage.generatePreSignedUploadUrl(BUCKET_NAME, OBJECT_KEY, OBJECT_CONTENT_TYPE);
+
+            // when
+            int actualStatusCode = storageClient.upload(fileInputStream, OBJECT_CONTENT_TYPE, preSignedUrl).getStatusCode();
+
+            // then
+            assertThat(actualStatusCode).isEqualTo(200);
+            List<String> actualBucketContent = testS3Storage.listBucketContent(BUCKET_NAME);
+            assertThat(actualBucketContent)
+                    .isNotNull()
+                    .containsOnly(OBJECT_KEY);
```

Comment:
I don't think you need the `isNotNull` here, and I think it would be more accurate to replace `containsOnly` with `isEqualTo` (since otherwise `[OBJECT_KEY, OBJECT_KEY]` will pass the test).